modules/openxOrtbBidAdapter.js

```diff
@@ -24,6 +22,99 @@ export const spec = {
 
+const converter = ortbConverter({
+  context: {
+    netRevenue: true,
+    ttl: 300
+  },
+  imp(buildImp, bidRequest, context) {
+    const imp = buildImp(bidRequest, context);
+    mergeDeep(imp, {
+      tagid: bidRequest.params.unit,
+      ext: {
+        divid: bidRequest.adUnitCode
+      }
+    });
+    if (bidRequest.params.customParams) {
+      utils.deepSetValue(imp, 'ext.customParams', bidRequest.params.customParams);
+    }
+    if (bidRequest.params.customFloor && !imp.bidfloor) {
+      imp.bidfloor = bidRequest.params.customFloor;
+    }
+    return imp;
+  },
+  request(buildRequest, imps, bidderRequest, context) {
+    const req = buildRequest(imps, bidderRequest, context);
+    mergeDeep(req, {
+      at: 1,
+      ext: {
+        bc: `${bidderConfig}_${bidderVersion}`
+      },
+      regs: {
+        // always set regs.coppa, even when not configured
+        // TODO: is this necessary?
+        coppa: config.getConfig('coppa') ? 1 : 0
+      }
+    })
+    const bid = context.bidRequests[0];
+    if (bid.params.doNotTrack) {
+      utils.deepSetValue(req, 'device.dnt', 1);
+    }
+    if (bid.params.platform) {
+      utils.deepSetValue(req, 'ext.platform', bid.params.platform);
+    }
+    if (bid.params.delDomain) {
+      utils.deepSetValue(req, 'ext.delDomain', bid.params.delDomain);
+    }
+    if (bid.params.test) {
+      req.test = 1
+    }
+    return req;
+  },
+  bidResponse(buildBidResponse, bid, context) {
+    const bidResponse = buildBidResponse(bid, context);
+    if (bid.ext) {
+      bidResponse.meta.networkId = bid.ext.dsp_id;
+      bidResponse.meta.advertiserId = bid.ext.buyer_id;
+      bidResponse.meta.brandId = bid.ext.brand_id;
+    }
+    const {ortbResponse} = context;
+    if (ortbResponse.ext && ortbResponse.ext.paf) {
+      // TODO: the following is likely incorrect when an ORTB response contains more than one bid
+      // all of them would be set with the same content_id (from the last bid), since they share the same `paf` object
+      bidResponse.meta.paf = ortbResponse.ext.paf;
+      bidResponse.meta.paf.content_id = utils.deepAccess(bid, 'ext.paf.content_id');
+    }
+    return bidResponse;
+  },
+  response(buildResponse, bidResponses, ortbResponse, context) {
+    // pass these from request to the responses for use in userSync
+    const {ortbRequest} = context;
+    if (ortbRequest.ext) {
+      if (ortbRequest.ext.delDomain) {
+        utils.deepSetValue(ortbResponse, 'ext.delDomain', ortbRequest.ext.delDomain);
+      }
+      if (ortbRequest.ext.platform) {
+        utils.deepSetValue(ortbResponse, 'ext.platform', ortbRequest.ext.platform);
+      }
+    }
+    return buildResponse(bidResponses, ortbResponse, context);
+  },
+  overrides: {
+    imp: {
+      bidfloor(setBidFloor, imp, bidRequest, context) {
+        // enforce floors should always be in USD
+        // TODO: does it make sense that request.cur can be any currency, but request.imp[].bidfloorcur must be USD?
```

Comment:
I believe the floors module defaults to USD anyway if getFloor is called without currency so I was just making it more clear. https://docs.prebid.org/dev-docs/modules/floors.html#currency. Also yes our exchange supports mismatched floor currencies.

Reply:
the base logic (that is overridden here) is to do - simplified:
```
  const {floor, currency} = bidRequest.getFloor({
    currency: getConfig('currency.adServerCurrency') || 'USD'
  )
  imp.bidfloor = floor;
  imp.bidfloorcur = currency;
```

this fails tests because they enforce that, if the currency returned by `getFloor` is not 'USD', the result from the floors module should be ignored.